File: backend-laravel/app/Models/JobExperience.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class JobExperience extends Model
{
    use HasFactory;

    protected $fillable = [
        'job_seeker_id',
        'title',
        'company',
        'employment_type',
        'location',
        'start_date',
        'end_date',
        'is_current',
        'description',
    ];

    protected $casts = [
        'is_current' => 'boolean',
        'start_date' => 'date:Y-m-d',
        'end_date' => 'date:Y-m-d',
    ];

    public function jobSeeker(): BelongsTo
    {
        return $this->belongsTo(JobSeeker::class);
    }
}

File: mobile-flutter/lib/models/job_experience.dart
class JobExperience {
  final int id;
  final int jobSeekerId;
  final String title;
  final String companyName;
  final String? location;
  final String? startDate;
  final String? endDate;
  final bool isCurrent;
  final String? description;

  JobExperience({
    required this.id,
    required this.jobSeekerId,
    required this.title,
    required this.companyName,
    this.location,
    this.startDate,
    this.endDate,
    this.isCurrent = false,
    this.description,
  });

  factory JobExperience.fromJson(Map<String, dynamic> json) {
    return JobExperience(
      id: json['id'] is int ? json['id'] : int.tryParse(json['id'].toString()) ?? 0,
      jobSeekerId: json['job_seeker_id'] is int
          ? json['job_seeker_id']
          : int.tryParse(json['job_seeker_id'].toString()) ?? 0,
      title: json['title'] ?? '',
      companyName: json['company_name'] ?? json['company'] ?? '',
      location: json['location'],
      startDate: json['start_date'],
      endDate: json['end_date'],
      isCurrent: json['is_current'] == true || json['is_current'] == 1 || json['is_current'] == '1',
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'job_seeker_id': jobSeekerId,
      'title': title,
      'company_name': companyName,
      'location': location,
      'start_date': startDate,
      'end_date': endDate,
      'is_current': isCurrent,
      'description': description,
    };
  }
}
